File: app/src/main/java/me/yangxiaobin/sample/aspectj/aspect/SampleAspect.java
package me.yangxiaobin.sample.aspectj.aspect;

import org.aspectj.lang.annotation.After;
import org.aspectj.lang.annotation.Aspect;

@Aspect
public class SampleAspect {

    @After("execution(public * *(..))")
    public void afterExecute () {
        System.out.println("---> sample aspect afterExecute ");
    }

}

File: androidapp/src/main/java/me/yangxiaobin/androidapp/aspect/SampleAspect.kt
package me.yangxiaobin.androidapp.aspect

import org.aspectj.lang.annotation.After
import org.aspectj.lang.annotation.Aspect


@Aspect
class SampleAspect {

    @After("execution(public * *(..))")
    fun doBefore() {
        println("-----> Im Before Aspect")
    }
}
